### DISCUSSION TITLE
iTwin Snapshot does not import custom schemas derived from dgn ItemType

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#5858] [Q&A]
When I used iTwin Synchronizer to sync a dgn file that contains elements with ItemTypes applied the resultant CONNECT i Model correctly creates ecschemas derived from these ItemTypes.  This can be verified both from imodelconsole and by selecting any of these elements from the CONNECT iTwin Viewer.

I found that if I take this same dgn file and create local bim file using iTwin Snapshot, the custom ecschemas are not created imported into the bim file and consequently the elements that would have used those ecclasses are not getting those ecclasses applied.

Is this a bug in iTwin Snapshot or expected behavior?

**Comment by JohnMajerle** [comment]
The original dgn file also came from OpenRoads.  I am finding that the snapshot bim file does not contain any Civil ecschemas.

**Comment by snehaInmdr** [comment]
Hi John, was the iTwin Synchronizer Client App launched from within a design application? Launching from within a design application helps resolve schemas. iTwin Snapshot cannot be launched in this way. Please share the logs for the iTwinSnapshot app so we can confirm the cause. Please click on gear icon on the top right and pick save logs from the menu and send us the saved log file.

**Comment by JohnMajerle** [comment]
@snehaInmdr - Thank you for the suggestion.  I found that when I ran iTwin Snapshot again the resultant bim file is now correct.  

There are a couple of possibilities for my previous problem.  I noticed that running iTwin Synchronizer triggered a pull of fresh plugins which may have fixed it.  It's also possible that my bad bim file was instead created with DgnV8Converter.exe which I know is no longer supported, which is unfortunate since iTwin Synchronizer does not  support a command line interface.  

Thanks for looking into this for me.
